core/app/models/concerns/spree/user_api_authentication.rb

```diff
@@ -13,7 +13,10 @@ def clear_spree_api_key!
     private
 
     def generate_spree_api_key
-      SecureRandom.hex(24)
+      loop do
+        random_token = SecureRandom.urlsafe_base64
+        break random_token unless self.class.exists?(spree_api_key: random_token)
```

Comment:
Please check below code, doing the same but looks more readable to me somehow :)
_token = SecureRandom.urlsafe_base64
until self.class.exists?(spree_api_key: random_token) do
  _token = SecureRandom.urlsafe_base64
end
_token